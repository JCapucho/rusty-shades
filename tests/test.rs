use rusty_shades::build_naga_ir;
use std::fs::read_to_string;

#[test]
fn insta_test() {
    insta::glob!("*.rsh", |file| {
        let code = read_to_string(file).unwrap();
        let res = build_naga_ir(&code);
        insta::assert_ron_snapshot!(res);
    });
}
