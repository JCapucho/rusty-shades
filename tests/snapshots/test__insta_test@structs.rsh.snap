---
source: tests/test.rs
expression: res
input_file: tests/structs.rsh
---
Ok(Module(
  header: Header(
    version: (1, 0, 0),
    generator: 1920299892,
  ),
  types: [
    Type(
      name: None,
      inner: Scalar(
        kind: Sint,
        width: 4,
      ),
    ),
    Type(
      name: Some("TupleStruct"),
      inner: Struct(
        members: [
          StructMember(
            name: Some("0"),
            origin: Offset(0),
            ty: 1,
          ),
          StructMember(
            name: Some("1"),
            origin: Offset(4),
            ty: 1,
          ),
        ],
      ),
    ),
    Type(
      name: Some("NormalStruct"),
      inner: Struct(
        members: [
          StructMember(
            name: Some("a"),
            origin: Offset(0),
            ty: 1,
          ),
          StructMember(
            name: Some("b"),
            origin: Offset(4),
            ty: 1,
          ),
        ],
      ),
    ),
  ],
  constants: [],
  global_variables: [],
  functions: [],
  entry_points: {},
))
